python/src/aiconfig/default_parsers/openai.py

```diff
@@ -156,6 +156,25 @@ async def serialize(
                     outputs=assistant_output,
                 )
                 prompts.append(prompt)
+            else:
```

Comment:
nit: do an explicit check -- elif i==0 and assistant=true

Reply:
updated, added a check `role == "assistant"`

side note: our automated tests also caught this, which is great.